File: front end/GP_project/src/api/reportlist.ts
import request from "@/utils/request"
import { c } from "vite/dist/node/types.d-FdqQ54oU";

//定义类型
export type Common<T> =  {
    count:string;
    code:string
    data:T
    mesg:string
}
export type reportItem = {
    id:number
    type:string
    theme:string
    reporter:string
    reportdate:string
    reportCode:string
    flightNumber:string
    reportIntroduction:string
    eventType:string
    eventRisk:string
    evaluationDate:string   //查看表有
    evaluator:string   //查看表有
    conclusion:string   //查看表有
    needReaserch:string
    needExpertResearch:string
    fileId:number
}

export type fileItem = {
    id:number
    fileName:string
    filePath:string
    fileSize:number
    fileType:string
    originName:string
}

export type CreateOrEditReportItem  = Omit<reportItem,'conclusion'| 'evaluationDate' | 'evaluator' > &{id?:number}
//请求列表数据
export const getReportList = (page:number) =>{
    return request<{code:number;count:number;msg:string;data:reportItem[]}>({
        method: 'GET',
        url:`/reportlist/list?page=${page}&size=15`,
    })
}

//上传
export const saveOrUpdate =(reportinfo :reportItem |CreateOrEditReportItem) =>{
    return request<{code:number;msg:string;data:reportItem[]}>({
        method: 'PUT',
        url:'/reportlist/',
        data:reportinfo
    })
}
//根据删除
export const deleteReport = (id:number) => {
    return request<{code:number;msg:string;data:reportItem[]}>({
        method: 'DELETE',
        url:`/reportlist/${id}`, //有参数，用``符，且不能有空格
    })
}
    //根据id查找
    export const getReportById = (id:string) =>{
        return request<{code:number;msg:string;data:reportItem}>({
            method: 'GET',
            url:`/reportlist/${id}`,
            params:{
                id,
            }
        })
    }

//根据id下载
export const download = async (id: number)=> {
    const response = await request({
      method: 'GET',
      url: `/file/download?id=${id}`,
      params: {
        id: id,
      },
      responseType: 'blob',
    });
    console.log("打印请求的东西",response.data)
    return response.data as Blob;
  };
//根据id找文件类型
export const getFile = async (id:string)=>{
    return request<{code:number;msg:string;data:fileItem}>({
        method: 'GET',
        url:`/file/${id}`,
        params:{
            id,
        }
    })
}

File: back end/quarkus_test1/src/main/java/com/example/myproject/entity/ReportList.java
package com.example.myproject.entity;


import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@Table(name = "reportlist")
public class ReportList implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "file_id")
    private String fileId;
    
    @Column
    private String type;
    
    @Column(name = "report_code")
    private String reportCode;
    
    @Column
    private String theme; 
    
    @Column
    private String reportdate;
    
    @Column
    private String reporter;
    
    
    @Column(name = "flight_number")
    private String flightNumber;
    
    
    @Column(name = "report_introduction")
    private String reportIntroduction;
    
    
    @Column(name = "event_type")
    private String eventType;
    
    
    @Column(name = "event_risk")
    private String eventRisk;
    
    @Column(name = "evaluation_date")
    private String evaluationDate;
    
    @Column
    private String evaluator;
    
    
    @Column
    private String conclusion;
    
    
    @Column(name = "need_research")
    private String needReaserch;
    
    @Column(name = "need_expert_research")
    private String needExpertResearch;
    
    
}
